File: backend/app/schemas/notification.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.models.notification import NotificationCategory, NotificationPriority


class NotificationResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    title: str
    message: str
    category: NotificationCategory
    priority: NotificationPriority
    is_read: bool
    related_id: Optional[str] = None
    related_url: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    total: int
    unread_count: int


class NotificationCreate(BaseModel):
    user_id: uuid.UUID
    title: str
    message: str
    category: NotificationCategory
    priority: NotificationPriority = NotificationPriority.MEDIUM
    related_id: Optional[str] = None
    related_url: Optional[str] = None


class NotificationMarkRead(BaseModel):
    notification_ids: list[uuid.UUID]

File: frontend/src/types/notification.ts
export type NotificationCategory = 'incident' | 'service_request' | 'change_request' | 'system'
export type NotificationPriority = 'low' | 'medium' | 'high'

export interface Notification {
  id: string
  user_id: string
  title: string
  message: string
  category: NotificationCategory
  priority: NotificationPriority
  is_read: boolean
  related_id: string | null
  related_url: string | null
  created_at: string
}

export interface NotificationListResponse {
  items: Notification[]
  total: number
  unread_count: number
}

export interface MarkReadRequest {
  notification_ids: string[]
}
